File: CS/WinSample.Module/Updater.cs
using System;

using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Reports;
using DevExpress.ExpressApp;

namespace WinSample.Module
{
    public class Updater : ModuleUpdater
    {
        public Updater(DevExpress.ExpressApp.IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            if (new XPCollection<DomainObject1>(Session).Count == 0) {
                double i = 1;
                double inc = 1.0 / 20.0;
                while (i < 3) {
                    DomainObject1 do1 = new DomainObject1(Session);
                    i += inc;
                    do1.Property1 = (int)i;
                    i += inc;
                    do1.Property2 = (int)i;
                    i += inc;
                    do1.Property3 = (int)i;
                    i += inc;
                    do1.Property4 = (int)i;
                    i += inc;
                    do1.Property5 = (int)i;
                    i += inc;
                    do1.Property6 = (int)i;
                    do1.Save();
                }
            }
            CreateReport("XafReport");
        }
        private void CreateReport(string reportName) {
            ReportData reportdata = Session.FindObject<ReportData>(new BinaryOperator("Name", reportName));
            if (reportdata == null) {
                reportdata = new ReportData(Session);
                XafReport rep = new XafReport();
                rep.ReportName = reportName;
                rep.ObjectSpace = new ObjectSpace(new UnitOfWork(Session.DataLayer), XafTypesInfo.Instance);
                rep.LoadLayout(GetType().Assembly.GetManifestResourceStream(
                   "WinSample.Module.Reports." + reportName + ".repx"));
                reportdata.SaveXtraReport(rep);
                reportdata.Save();
            }
        }
    }
}

File: VB/WinSample.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Reports
Imports DevExpress.ExpressApp

Namespace WinSample.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As DevExpress.ExpressApp.IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			If ObjectSpace.GetObjects(Of DomainObject1)().Count = 0 Then
				Dim i As Double = 1
				Dim inc As Double = 1.0 / 20.0
				Do While i < 3
					Dim do1 As DomainObject1 = ObjectSpace.CreateObject(Of DomainObject1)()
					i += inc
					do1.Property1 = CInt(Fix(i))
					i += inc
					do1.Property2 = CInt(Fix(i))
					i += inc
					do1.Property3 = CInt(Fix(i))
					i += inc
					do1.Property4 = CInt(Fix(i))
					i += inc
					do1.Property5 = CInt(Fix(i))
					i += inc
					do1.Property6 = CInt(Fix(i))
					do1.Save()
				Loop
			End If
			CreateReport("XafReport")
		End Sub
		Private Sub CreateReport(ByVal reportName As String)
			Dim reportdata As ReportData = ObjectSpace.FindObject(Of ReportData)(New BinaryOperator("Name", reportName))
			If reportdata Is Nothing Then
				reportdata = ObjectSpace.CreateObject(Of ReportData)()
				Dim rep As New XafReport()
				rep.ObjectSpace = ObjectSpace
				rep.LoadLayout(Me.GetType().Assembly.GetManifestResourceStream("Reports." & reportName & ".repx"))
				rep.ReportName = reportName
				reportdata.SaveReport(rep)
			End If

		End Sub
	End Class
End Namespace
